src/Bicep.Core/FileSystem/PathHelper.cs

```diff
@@ -94,6 +98,17 @@ public static string GetDefaultBuildOutputPath(string path)
             return Path.ChangeExtension(path, TemplateOutputExtension);
         }
 
+        public static string GetDefaultParamBuildOutputPath(string path)
+        {
+            if (string.Equals(Path.GetExtension(path), ParamsOutputExtension, PathComparison))
+            {
+                // throwing because this could lead to us destroying the input file if extensions get mixed up.
+                throw new ArgumentException($"The specified file already has the '{TemplateOutputExtension}' extension.");
+            }
+
+            return Path.ChangeExtension(path, ParamsOutputExtension);
+        }
```

Comment:
Why do we need this logic? Shouldn't we be able to just use `GetDefaultBuildOutputPath`?

Reply:
Because the default is building to main.json which is used for bicep file itself and I am trying to build to main.bicepparam.json